google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java

```diff
@@ -0,0 +1,282 @@
+            .setUserIdentifierSource(UserIdentifierSource.FIRST_PARTY)
+            .setAddressInfo(
+                OfflineUserAddressInfo.newBuilder()
+                    .setHashedFirstName(normalizeAndHash(sha256Digest, "Joanna"))
+                    .setHashedLastName(normalizeAndHash(sha256Digest, "Smith"))
+                    .setHashedStreetAddress(
+                        normalizeAndHash(sha256Digest, "1600 Amphitheatre Pkwy"))
+                    .setCity("Mountain View")
+                    .setState("CA")
+                    .setPostalCode("94043")
+                    .setCountryCode("US"))
+            .build();
+
+    // Creates a user identifier using the hashed email address.
+    UserIdentifier emailIdentifier =
+        UserIdentifier.newBuilder()
+            .setUserIdentifierSource(UserIdentifierSource.FIRST_PARTY)
+            // Uses the normalize and hash method specifically for email addresses.
+            .setHashedEmail(normalizeAndHashEmailAddress(sha256Digest, "joannasmith@example.com"))
+            .build();
+
+    // Adds the user identifiers to the enhancement adjustment.
+    enhancementBuilder.addUserIdentifiers(addressIdentifier).addUserIdentifiers(emailIdentifier);
+
+    // Sets optional fields where a value was provided.
+
+    if (userAgent != null) {
+      // Sets the user agent. This should match the user agent of the request that sent the original
+      // conversion so the conversion and its enhancement are either both attributed as same-device
+      // or both attributed as cross-device.
+      enhancementBuilder.setUserAgent(userAgent);
+    }
+
+    if (restatementValue != null) {
+      // Creates a builder to construct the restated conversion value.
+      RestatementValue.Builder valueBuilder = enhancementBuilder.getRestatementValueBuilder();
+      // Sets the new value of the conversion.
+      valueBuilder.setAdjustedValue(restatementValue);
+      // Sets the currency of the new value, if provided. Otherwise, the default currency from
+      // the conversion action is used, and if that is not set then the account currency is used.
+      if (restatementCurrencyCode != null) {
+        valueBuilder.setCurrencyCode(restatementCurrencyCode);
+      }
+    }
+    // [END create_adjustment]
+
+    // Creates the conversion upload service client.
+    try (ConversionAdjustmentUploadServiceClient conversionUploadServiceClient =
+        googleAdsClient.getLatestVersion().createConversionAdjustmentUploadServiceClient()) {
+      // Uploads the enhancement adjustment. Partial failure should always be set to true.
+      UploadConversionAdjustmentsResponse response =
+          conversionUploadServiceClient.uploadConversionAdjustments(
+              UploadConversionAdjustmentsRequest.newBuilder()
+                  .setCustomerId(Long.toString(customerId))
+                  .addConversionAdjustments(enhancementBuilder)
```

Comment:
It doesn't look like enhancementBuilder builder was ever actually built. Is this correct? Does it work?